config/v1/types_console.go

```diff
@@ -26,6 +28,26 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// customization is used to optionally provide a small set of
+	// customization options to the web console.
+	// +optional
+	Customization ConsoleCustomization `json:"customization,omitempty"`
+}
+
+// ConsoleCustomization defines a list of optional configuration for the console.
+type ConsoleCustomization struct {
+	// customURL contains URL and certificate for replacing the default console URL.
+	// +optional
+	CustomURL `json:customURL,omitempty`
+}
+
+// CustomURL replaces the default console URL under which console will be available.
+type CustomURL struct {
+	// host is the desired custom URL under which console will be available.
+	Host string `json:host:omitempty`
+	// servingCert references to ConfigMap in the openshift-config namespace that
+	// contains custom certificate.
+	ServingCert ConfigMapNameReference `json:servingCert:omitempty`
```

Comment:
Wasn't really sure about this name, but didn't want to duplicate the `custom*` prefix in every field...

Reply:
given it also needs to contain the private key (you can't serve a cert without a private key) shouldn't this reference a secret?

Reply:
Just update the PR with the `SecretReference`. Thanks !